### PR TITLE
[PAGOPA-2446] refactoring: Update status on-the-fly

### PR DESCRIPTION
**Pull request by cap-ang**
#### List of Changes

TL;DR
- `validityCheckAndUpdate` method has been added 🆕  -> It is used when Get and Pay Payment Option
    
- `expirationCheckAndUpdate` method has been added 🆕 -> It is used **only** when Get Payment Option
     
- Check on expiration or validity on report API has been removed 🗑️ 

These scenarios were refactored:

[GET PO by NAV scenario]
- check validity (if published and now > validityDate) -> update
- check expiration (if switchToExpired and valid and  > now) -> update

[PAY scenario]
- check validity (if published and now > validityDate) -> update
- expiration check is useless (pay already done)

[INVALIDATE scenario]
- check expiration because expired isn't an invalidable status
- validity check is useless (both published and valid are invalidable status)

#### Motivation and Context

Refactoring status retrieve _on the fly_  handling. **Cron** granularity differs from `validityDate` and `dueDate` granularity. These changes helps to update status on-the-fly.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by pasqualespica — APPROVED** [review]
great work 🚀
